extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java

```diff
@@ -159,7 +162,8 @@ RouteBuildItem handler(LaunchModeBuildItem launch,
             recorder.setupClDevMode(shutdownContext);
             displayableEndpoints.produce(new NotFoundPageDisplayableEndpointBuildItem(openApiConfig.path));
         }
-        return new RouteBuildItem(openApiConfig.path, new OpenApiHandler(), HandlerType.BLOCKING);
+        OpenApiHandler openApiHandler = new OpenApiHandler(openApiRuntimeConfig);
```

Comment:
It would be better if the handler was created by the recorder. And you could use the same technique I mention below returning two different handlers depending on the config.

Reply:
Done